sklearn/ensemble/_gb.py

```diff
@@ -653,7 +890,9 @@ def _fit_stages(
             if self.n_iter_no_change is not None:
                 # By calling next(y_val_pred_iter), we get the predictions
                 # for X_val after the addition of the current stage
-                validation_loss = loss_(y_val, next(y_val_pred_iter), sample_weight_val)
+                validation_loss = self._loss(
```

Comment:
Does this loss require a factor?

Reply:
Yes, but almost no. The factor in this place does not change the behavior / program execution and the `validation_loss` is not made public anywhere.
Nevertheless, I added it.